services/app-web/src/documentHelpers/makeDocumentKeyLookupList.ts

```diff
@@ -0,0 +1,50 @@
+import { base64ToDomain } from '../common-code/proto/healthPlanFormDataProto';
+import { HealthPlanPackage } from '../gen/gqlClient'
+import { parseKey } from '../common-code/s3URLEncoding'
+
+type LookupListType = {
+    currentDocuments: string[]
+    previousDocuments: string[]
+}
+
+const getKey = (s3URL: string) => {
+    const key = parseKey(s3URL)
+    return key instanceof Error ? null : key
+}
+
+export const makeDocumentList = (
+    submissions: HealthPlanPackage
+): LookupListType | Error => {
+    const docBuckets = [
+        'contractDocuments',
+        'rateDocuments',
+        'documents',
+    ] as const
+    const lookupList: LookupListType = {
+        currentDocuments: [],
+        previousDocuments: [],
+    }
+
+    const revisions = submissions.revisions
+
+    for (let index = 0; index < revisions.length; index++) {
+        const revisionData = base64ToDomain(revisions[index].node.formDataProto)
+        if (revisionData instanceof Error) {
+            return new Error(
```

Comment:
>I’m curious about this if, too, there’s no harm in calling .forEach() on an empty array.

>In general I’ve been prefering the for (const revision of revisions) {} syntax, but I don’t know if you can get an index out of that. I think it might be a little more elegant to make a separate function that takes a revision and an array to dump its documents into and call that differently for the first and then the rest of the reivisions.

@macrael I switched to the vanilla for loop, returning errors now breaks out of the loop! I also added a few test, one which tests when the functions returns an error.